crates/accelerate/src/disjoint_layout.rs

```diff
@@ -0,0 +1,488 @@
+                };
+                let raw_blocks = py_inst.instruction.getattr(py, "blocks").unwrap();
+                let blocks: &Bound<PyTuple> = raw_blocks.downcast_bound::<PyTuple>(py).unwrap();
+                for block in blocks.iter() {
+                    let mut inner_wire_map = vec![Qubit(u32::MAX); wire_map.len()];
+                    let node_qargs = dag.get_qargs(inst.qubits);
+
+                    for (outer, inner) in node_qargs.iter().zip(0..inst.op.num_qubits()) {
+                        inner_wire_map[inner as usize] = wire_map[outer.index()]
+                    }
+                    let block_dag = circuit_to_dag(py, block.extract()?, false, None, None)?;
+                    build_interaction_graph(
+                        &block_dag,
+                        &inner_wire_map,
+                        im_graph,
+                        im_graph_node_map,
+                        reverse_im_graph_node_map,
+                    )?;
+                }
+                Ok(())
+            })?;
+            continue;
+        }
+        let len_args = inst.op.num_qubits();
+        if len_args == 1 {
+            let dag_qubits = dag.get_qargs(inst.qubits);
+            let qargs = wire_map[dag_qubits[0].index()];
+            if im_graph_node_map[qargs.index()].is_none() {
+                let new_index = im_graph.add_node(());
+                im_graph_node_map[qargs.index()] = Some(new_index);
+                reverse_im_graph_node_map[new_index.index()] = Some(qargs);
+            }
+        } else if len_args == 2 {
+            let dag_qubits = dag.get_qargs(inst.qubits);
+            let qargs: [Qubit; 2] = [
+                wire_map[dag_qubits[0].index()],
+                wire_map[dag_qubits[1].index()],
+            ];
+            if im_graph_node_map[qargs[0].index()].is_none() {
+                let new_index = im_graph.add_node(());
+                im_graph_node_map[qargs[0].index()] = Some(new_index);
+                reverse_im_graph_node_map[new_index.index()] = Some(qargs[0]);
+            }
+            if im_graph_node_map[qargs[1].index()].is_none() {
+                let new_index = im_graph.add_node(());
+                im_graph_node_map[qargs[1].index()] = Some(new_index);
+                reverse_im_graph_node_map[new_index.index()] = Some(qargs[1]);
+            }
+
+            let edge_index = im_graph.find_edge(
+                im_graph_node_map[qargs[0].index()].unwrap(),
+                im_graph_node_map[qargs[1].index()].unwrap(),
+            );
+            if edge_index.is_none() {
+                im_graph.add_edge(
+                    im_graph_node_map[qargs[0].index()].unwrap(),
+                    im_graph_node_map[qargs[1].index()].unwrap(),
+                    (),
+                );
+            }
+        }
+        if len_args > 2 {
+            return Err(MultiQEncountered::new_err(""));
+        }
```

Comment:
This is copied over from VF2, but in this situation, is it really an error during the disjoint handling to have a multi-q gate?  Feels like it should be up to the layout pass to decide that.